code/modules/mob/living/carbon/carbon_organs.dm

```diff
@@ -61,3 +59,19 @@
 	//Check if we should die
 	if(species.is_vital_organ(src, O))
 		death()
+
+//Called by surgeries when detaching an organ during organ detach surgery
+/mob/living/carbon/surgical_detach_organ(var/obj/item/organ/O, var/obj/item/organ/external/parent)
+	remove_organ(O, FALSE, TRUE)
+	O.forceMove(src)
+	LAZYDISTINCTADD(parent.implants, O.get_detached_organ()) //#FIXME: get_detached_organ() is a placeholder for handling the weird behavior of the mmi_holder
+
+//Called by surgery when placing an organ during organ attach surgery. Happens before attaching
+/mob/living/carbon/surgical_place_organ(var/obj/item/organ/O, var/obj/item/organ/external/parent)
+	LAZYDISTINCTADD(parent.implants, O)
+	O.forceMove(src)
+
+/mob/living/carbon/get_contained_external_atoms()
```

Comment:
This is redundant, human get_contained_external_atoms already subtracts the external organ list, which stumps should be in.

Reply:
It doesn't, because I had to put that there in order to prevent the stumps from coming out. I couldn't find another implementation.

Reply:
Actually, after thinking about it, I know what's up. Stumps were getting installed on initialize before.. Going to fix that.

Reply:
Nah, they still get dumped. Gonna need to mess with the gib proc methinks.

Reply:
````
/mob/living/carbon/get_contained_external_atoms()
	. = ..()
	LAZYREMOVE(., get_organs())
````
Current `dev`, `living.dm`.

Reply:
> ```
> /mob/living/carbon/get_contained_external_atoms()
> 	. = ..()
> 	LAZYREMOVE(., get_organs())
> ```
> 
> Current `dev`, `living.dm`.
Yeah, realized that earlier when I fixed stumps not getting added to the external_organ list.
But the line does nothing in our particular case. Might be because the organ list are cleared before dump_contents() is called in destroy. So they still get dumped anyways. Could also be the gib proc kicking them out the body on gib. But it shouldn't since stumps cannot be amputated.

Reply:
I added a is_droppable() proc to decide if an organ should be allowed to exist outside the mob. Otherwise it just gets deleted instead of dropped. Seems to be working well to prevent stumps and the root limb from just getting kicked out. 

Only weird thing I'm noticing is that gib() doesn't seem to run consistently. Sometimes everything gets kicked out like it should, some other times, the lower body organs don't get kicked out for some reasons. Not entirely sure what's happening there, since everything seems 100% fine structure wise. Its like gib is called in parallel with destruction almost.